vlm/src/main/scala/geotrellis/contrib/vlm/RasterSourceMetadata.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm
+
+import geotrellis.proj4.CRS
+import geotrellis.raster.{CellSize, CellType, GridExtent, RasterExtent}
+import geotrellis.vector.Extent
+
+trait RasterSourceMetadata {
```

Comment:
FTW! Wrapping all of the below together is fantastic. It allows me to get rid of [this hack](https://github.com/s22s/rasterframes/blob/develop/core/src/main/scala/org/locationtech/rasterframes/ref/SimpleRasterInfo.scala) which was an attempt to cache this info in a serializable format. Can't wait to use it.